orcid-scheduler-web/src/main/java/org/orcid/scheduler/loader/source/cli/RingoldToRorCSVMapping.java

```diff
@@ -0,0 +1,121 @@
+    @SuppressWarnings({ "resource" })
+    @PostConstruct
+    private void init() {
+        ApplicationContext context = new ClassPathXmlApplicationContext("orcid-scheduler-context.xml");
+        orgDisambiguatedDao = (OrgDisambiguatedDao) context.getBean("orgDisambiguatedDao");
+        orgDisambiguatedManager = (OrgDisambiguatedManager) context.getBean("orgDisambiguatedManager");
+    }
+
+    public static void main(String[] args) {
+        RingoldToRorCSVMapping mappingData = new RingoldToRorCSVMapping();
+        // TODO Auto-generated method stub
+        try {
+
+            mappingData.init();
+            mappingData.getCSVMapping();
+
+        } catch (Exception e) {
+            LOGGER.error("Exception when generating  csv mapping Ringgolds to Rors", e);
+            System.err.println(e.getMessage());
+        } finally {
+            System.exit(0);
+        }
+
+    }
+
+    public void getCSVMapping() {
+        LOGGER.info("About to start ringgold to ror csv mapping");
+        List<OrgDisambiguatedEntity> entities = null;
+        HashMap<String, String> ringgoldMap = new HashMap<String, String>();
+        ringgoldMap.put("Ringgold", "ROR");
+        int startIndex = 0;
+        do {
+            entities = orgDisambiguatedDao.findBySourceType(RINGGOLD_TYPE, startIndex, INDEXING_CHUNK_SIZE);
+            LOGGER.info("Found chunk of {} disambiguated orgs for indexing", entities.size());
+            for (OrgDisambiguatedEntity entity : entities) {
+                if (StringUtils.equals(entity.getStatus(), OrganizationStatus.PART_OF_GROUP.name())) {
+                    for (OrgDisambiguatedExternalIdentifierEntity externalIdentifier : entity.getExternalIdentifiers()) {
+                        if (StringUtils.equals(externalIdentifier.getIdentifierType(), FunderIdentifierType.ISNI.value())) {
+                            List<OrgDisambiguated> orgsFromExternalIdentifier = orgDisambiguatedManager
+                                    .findOrgDisambiguatedIdsForSameExternalIdentifier(externalIdentifier.getIdentifier(), FunderIdentifierType.ISNI.value());
+                            if (orgsFromExternalIdentifier != null) {
+                                orgsFromExternalIdentifier.stream().forEach((o -> {
+
+                                    if (o.getSourceType().equals(OrgDisambiguatedSourceType.ROR.name())) {
+                                        ringgoldMap.put(entity.getSourceId(), o.getSourceId());
+                                    }
+
+                                }));
+                            }
+                        } else if(StringUtils.equals(externalIdentifier.getIdentifierType(), OrgDisambiguatedSourceType.ROR.name())) {
+                            ringgoldMap.put(entity.getSourceId(),externalIdentifier.getIdentifier());
+                        }
+                    }
+                    
+                  //TBD check if there is any ROR that has the ringgold as external identifier??
```

Comment:
@Camelia-Orcid there is just one case for this: 

```
orcid=# select e.id, e.org_disambiguated_id, d.source_type, d.name from org_disambiguated_external_identifier e, org_disambiguated d where e.identifier_type = 'RINGGOLD' an
d e.org_disambiguated_id = d.id;
   id    | org_disambiguated_id | source_type |                     name
---------+----------------------+-------------+----------------------------------------------
  989734 |              2115585 | GRID        | State University of Economics and Technology
 1000541 |              2132353 | GRID        | PNPE Company (Iran)
 1344079 |              2238571 | ROR         | State University of Economics and Technology
 1462954 |              2243764 | ROR         | PNPE Company (Iran)
(4 rows)
```

I think you can manually add this to the CSV, OK?